### DISCUSSION TITLE
Collection attributes mapping to Spline UI

### DISCUSSION THREAD

**Discussion by abhineet13** [AbsaOSS/spline#1045] [Q&A]
## Background [Optional]
Hi
We are writing AQLs on Arango Spline db to read the execution plan in tabular format and could not find certain fields like 
1.) "Type" (JDBC)
2.) "Destination" of "type" for jdbc,  jdbc string shows on UI but could not figure which collection has this information.
3.) "Executed at" on spline UI

## Question
Is there a document mapping Arango collection attributes to Spline UI.

AQL:
```aql
FOR a in operation
    FOR doc IN executionPlan 
        FOR b in progressOf
            FOR c in progress
                FILTER (DATE_FORMAT(DATE_ISO8601(a._created),"%yyyy-%mm-%dd")=='2022-03-21'
                    OR  DATE_FORMAT(DATE_ISO8601(a._created),"%yyyy-%mm-%dd")=='2022-03-22')
                   AND doc._id==a._belongsTo
                   AND doc._id==b._to
                   AND b._from==c._id
                SORT a._id desc
                RETURN DISTINCT {
                    "datev"         : DATE_FORMAT(DATE_FORMAT(DATE_ISO8601(a._created),"%yyyy-%mm-%dd %hh:%mm:%ss"),"%yyyy-%mm-%dd"),
                    "appName"       : doc.name,
                    "sourceType"    : a.extra.sourceType,
                    "inputSources"  : a.inputSources,
                    "appId"         : c.extra.appId,
                    "id"            : doc._id, 
                    "operation_name": a.name,
                    "operation_id"  : execution._id,
                    "writeCnt"      : c.extra.writeMetrics.numOutputRows,
                    "duration"      : c.durationNs,
                    "param"         : a.params
                } 
```

Thanks
Abhineet

**Comment by wajda** [comment]
**Suggestion:**
I'm not sure I understand semantics of your query, but from the performance perspective, it looks quite unnecessary expensive.
It gives you all permutations of _operation_, _execution plan_ and _progress_ triplets, filtered by the operation entity (outer most loop) with the filter expression itself sitting in the inner most loop. That leads to a big number of unnecessary iterations over 3 entities (`executionPlan`, `progressOf` and `progress`) before the whole path is rejected by the filter. If you brake your conjunctive filter expression  into separate `FILTER` statements if will allow for moving filters up optimization and early pruning of the outer cost iteration, potentially speeding the query up multiple times.

**Regarding you question**, there is a [persistent model diagram](https://github.com/AbsaOSS/spline/blob/gh-pages/docs/Persistent%20model.pdf) describing the meaning and relations between entities. Hope it helps you. 
Chronology related properties (like "executed at") are located in the `progress` entity, while the destination source and type related ones are located in the respective read/write `operation`s
